Libraries/Components/CheckBox/CheckBox.android.js

```diff
@@ -80,81 +135,59 @@ type DefaultProps = {
  * @keyword checkbox
  * @keyword toggle
  */
-let CheckBox = createReactClass({
-  displayName: 'CheckBox',
-  propTypes: {
-    ...DeprecatedViewPropTypes,
-    /**
-     * The value of the checkbox.  If true the checkbox will be turned on.
-     * Default value is false.
-     */
-    value: PropTypes.bool,
-    /**
-     * If true the user won't be able to toggle the checkbox.
-     * Default value is false.
-     */
-    disabled: PropTypes.bool,
-    /**
-     * Used in case the props change removes the component.
-     */
-    onChange: PropTypes.func,
-    /**
-     * Invoked with the new value when the value changes.
-     */
-    onValueChange: PropTypes.func,
-    /**
-     * Used to locate this view in end-to-end tests.
-     */
-    testID: PropTypes.string,
-  },
+class CheckBox extends React.Component<Props> {
+  _nativeRef: ?React.ElementRef<CheckBoxNativeType> = null;
 
-  getDefaultProps: function(): DefaultProps {
-    return {
-      value: false,
-      disabled: false,
-    };
-  },
+  static defaultProps = {
+    value: false,
+    disabled: false,
+  };
+
+  _setAndForwardRef = nativeRef => {
+    const {ref} = this.props;
 
-  mixins: [NativeMethodsMixin],
+    this._nativeRef = nativeRef;
 
-  _rctCheckBox: {},
-  _onChange: function(event: Object) {
-    this._rctCheckBox.setNativeProps({value: this.props.value});
+    // Forward to user ref prop (if one has been specified)
+    // String-based refs cannot be shared.
+    if (typeof ref === 'function') {
+      ref(nativeRef);
+    } else if (typeof ref === 'object' && ref != null) {
+      ref.current = nativeRef;
+    }
+  };
+
+  _onChange = (event: CheckBoxEvent) => {
+    nullthrows(this._nativeRef).setNativeProps({value: this.props.value});
```

Comment:
Since the original code doesn't have `nullthrows`, I'm not sure if we should introduce it here.